ThunderCats/UIColor+TCAdditions.m

```diff
@@ -28,12 +28,19 @@
 
 
 #import "UIColor+TCAdditions.h"
+#import "TCInvalidArgument.h"
 
 @implementation UIColor (TCAdditions)
 
 + (UIColor *)tc_colorWithHexString:(NSString *)hexString
 {
     NSString *cleanString = [hexString stringByReplacingOccurrencesOfString:@"#" withString:@""];
+    
+    if ([cleanString length] != 3 && [cleanString length] != 6)
+    {
+        [TCInvalidArgument raiseWithReason:@"invalid hex string"];
+    }
+    
```

Comment:
We also need to make sure that the string only contains valid hexadecimal characters.  The digits 0-9, and the letters a-f (upper or lower case are both fine).

And arguably, even if we add this check in, we'd still be letting a string like `##0#123#4#5##` through because we just strip the `#` symbols out.  And I'd argue that this should return a valid color.